src/systems/drawing-menu-system.js

```diff
@@ -0,0 +1,200 @@
+      const userinput = AFRAME.scenes[0].systems.userinput;
+
+      for (let i = 0; i < remotes.length; i++) {
+        const remote = remotes[i];
+        if (interaction.state[remote]) {
+          const hovered = interaction.state[remote].hovered;
+          if (
+            hovered &&
+            interaction.options[remote] &&
+            userinput.get(interaction.options[remote].grabPath) &&
+            hovered.components.tags &&
+            hovered.components.tags.data.singleActionButton &&
+            hovered.parentEl.parentEl.components["networked-drawing"]
+          ) {
+            if (hovered.classList.contains("undo-drawing")) {
+              hovered.parentEl.parentEl.components["networked-drawing"].undoDraw();
+            } else if (hovered.classList.contains("delete-drawing")) {
+              NAF.utils.takeOwnership(hovered.parentEl.parentEl);
+              this.sceneEl.removeChild(hovered.parentEl.parentEl);
+            }
+          }
+        }
+      }
+    };
+  })();
+
+  getHovered() {
+    const interaction = AFRAME.scenes[0].systems.interaction;
+    let hovered = null;
+    if (
+      interaction.state.leftRemote &&
+      interaction.state.leftRemote.hovered &&
+      interaction.state.leftRemote.hovered.classList.contains("drawing")
+    ) {
+      hovered = interaction.state.leftRemote.hovered;
+    } else if (
+      interaction.state.rightRemote &&
+      interaction.state.rightRemote.hovered &&
+      interaction.state.rightRemote.hovered.classList.contains("drawing")
+    ) {
+      hovered = interaction.state.rightRemote.hovered;
+    }
+    return hovered;
+  }
+
+  showMenu = (() => {
+    const position = new THREE.Vector3();
+    const cameraWorldPos = new THREE.Vector3();
+    return function(hovered, intersectionPoint) {
+      const menu = hovered.children[0];
+
+      if (!menu.object3D.visible || !almostEquals(0.25, intersectionPoint, this.lastIntersection)) {
+        this.camera.object3D.getWorldPosition(cameraWorldPos);
+        position
+          .subVectors(cameraWorldPos, intersectionPoint)
+          .normalize()
+          .multiplyScalar(0.1);
+        menu.object3D.position.copy(intersectionPoint).add(position);
+        menu.object3D.lookAt(cameraWorldPos);
+        menu.object3D.matrixNeedsUpdate = true;
+        menu.object3D.visible = true;
+        this.lastIntersection.copy(intersectionPoint);
+
+        const isMine = hovered.components.networked.isMine();
+        if (!isMine) {
+          menu.querySelector(".undo-drawing").object3D.visible = false;
+          const deleteButton = menu.querySelector(".delete-drawing");
+          deleteButton.object3D.position.y = 0.125;
+          deleteButton.object3D.matrixNeedsUpdate = true;
+        }
+
+        const dist = cameraWorldPos.distanceTo(menu.object3D.position);
+        const finalScale = this.getMenuScale(dist);
+
+        menu.removeAttribute("animation__show");
+
+        menu.setAttribute("animation__show", {
```

Comment:
Nice I keep forgetting that we can add animations to the interactions. Would like to add juice like this more often!